server/chat-plugins/info.js

```diff
@@ -2292,7 +2292,19 @@ const commands = {
 		if (!target) return this.parse('/help htmlbox');
 		target = this.canHTML(target);
 		if (!target) return;
-
+		if (target.includes('\n')) {
+			let OpenTags = 0;
+			let CloseTags = 0;
+			for (let line of target.split('\n')) {
+				OpenTags = OpenTags + (line.match(/<[^/].*>/g) || []).length;
+				CloseTags = CloseTags + (line.match(/<\/.*>/g) || []).length;
+				if (OpenTags > CloseTags) {
+					target = target.replace('\n', ' ')
+				} else if (OpenTags === CloseTags) {
+					target = target.replace('\n', '&#10;')
```

Comment:
Should I use ternaries here? I feel that nested ternaries are a bit unreadable

Reply:
You wouldn't need to nest them.

Also this isn't very DRY; you might want to write a function for this.

Reply:
I'd implement this as a few regexes.

Reply:
```ts
target = target.replace(/<[^>]*>/g, tag => tag.replace(/\n/g, ' '));
target = target.replace('\n', '&#10;');
```

This probably belongs as a help function in `Chat`, rather than duplicated five places here.

Reply:
Care to explain why you don't replace `\n` globally in the last line (or was it mistake in your end), because some tags will end up on a linebreak, if the right criteria is met.

Reply:
Oh that was a typo. Should be:

```js
target = target.replace(/<[^>]*>/g, tag => tag.replace(/\n/g, ' '));
target = target.replace(/\n/g, '&#10;');
```